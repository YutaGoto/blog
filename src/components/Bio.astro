---
import profile from "../../public/profile.json"
---

<div
  id="bio"
  class="mb-2 bg-gray-50 p-5 shadow-lg dark:bg-gray-800 sm:rounded-xl md:max-w-screen-2xl lg:max-w-sm"
  transition:name="bio"
  transition:animate="fade"
>
  <div class="max-w-fit">
    <img
      class="bio-avatar h-48 w-48 rounded-full object-contain"
      src="/blog/assets/images/profile-pic.png"
      alt="ProfilePicture"
    />

    <div class="flex-none space-y-3 pt-3">
      <div>
        <p class="mb-2 text-3xl">{profile.author.name}</p>
        <p class="break-words text-xl">{profile.author.summary}</p>
      </div>
      <div class="flex flex-1 space-x-2">
        {
          profile.social?.map(s => (
            <a
              href={s.url}
              class="object-bottom"
              target="_blank"
              rel="noreferrer noopener"
            >
              <i class={`ri-${s?.name}-fill ri-3xl`} />
            </a>
          ))
        }
      </div>
    </div>
  </div>
</div>
