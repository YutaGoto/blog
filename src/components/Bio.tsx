import React, { FC } from "react"
import { graphql, useStaticQuery } from "gatsby"
import { ImageFormat, StaticImage } from "gatsby-plugin-image"
import "./fontawesome"

export const Bio: FC = () => {
  const data = useStaticQuery<GatsbyTypes.BioQuery>(
    graphql`
      query Bio {
        site {
          siteMetadata {
            author {
              name
              summary
            }
            social {
              name
              url
            }
          }
        }
      }
    `
  )

  const author = data.site?.siteMetadata?.author
  const social = data.site?.siteMetadata?.social
  const formats: ImageFormat[] = ["auto", "webp", "avif"]

  return (
    <div className="px-4 py-10 bg-gray-50 dark:bg-gray-800 shadow-lg sm:rounded-xl sm:p-5">
      <div className="flex space-x-4">
        <div className="flex-none">
          <StaticImage
            className="bio-avatar"
            layout="fixed"
            formats={formats}
            src="../images/profile-pic.png"
            width={180}
            height={180}
            quality={95}
            alt="Profile picture"
          />
        </div>

        <div className="flex-1 flex flex-col space-y-3">
          <div>
            <p className="text-3xl mb-2">{author?.name}</p>
            <p className="text-xl">{author?.summary}</p>
          </div>
          <div className="flex-1 flex space-x-6">
            {social?.map(s => (
              <a
                key={s?.name}
                href={s?.url}
                className="object-bottom"
                area-label={s?.name}
                target="_blank"
                rel="noreferrer noopener"
              >
                <span className="icon">
                  <i
                    className={`fab fa-${s?.name} fa-lg`}
                    aria-hidden="true"
                  ></i>
                </span>
              </a>
            ))}
          </div>
        </div>
      </div>
    </div>
  )
}
