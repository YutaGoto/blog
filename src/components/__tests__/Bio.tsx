import React from "react"
import renderer from "react-test-renderer"
import { render } from "@testing-library/react"
import { PureBio as Bio } from "../Bio"

describe("bio", () => {
  const data: GatsbyTypes.BioQuery = {
    site: {
      siteMetadata: {
        author: {
          name: `Yuta Goto`,
          summary: `エンジニアです。`,
        },
        social: [
          {
            name: `twitter`,
            url: `https://twitter.com/gggooottto`,
          },
          {
            name: `github`,
            url: `https://github.com/YutaGoto`,
          },
        ],
      },
    },
  }

  it("renders correctly", () => {
    const tree = renderer.create(<Bio data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("display the div element", () => {
    const { container } = render(<Bio data={data} />)
    expect(container.querySelector(`div`)).toBeInTheDocument()
  })
})
