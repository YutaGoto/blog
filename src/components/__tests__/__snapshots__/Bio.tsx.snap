// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bio renders correctly 1`] = `
<div
  className="mb-2 bg-gray-50 p-5 shadow-lg dark:bg-gray-800 sm:rounded-xl "
>
  <div
    className="flex space-x-4"
  >
    <div
      className="flex-none"
    />
    <div
      className="flex flex-1 flex-col space-y-3"
    >
      <div>
        <p
          className="mb-2 text-3xl"
        >
          Yuta Goto
        </p>
        <p
          className="text-xl"
        >
          エンジニアです。
        </p>
      </div>
      <div
        className="flex flex-1 space-x-6"
      >
        <a
          className="object-bottom"
          href="https://twitter.com/gggooottto"
          rel="noreferrer noopener"
          target="_blank"
        >
          <span
            className="icon"
          >
            <i
              aria-hidden="true"
              className="fab fa-twitter fa-lg"
            />
          </span>
        </a>
        <a
          className="object-bottom"
          href="https://github.com/YutaGoto"
          rel="noreferrer noopener"
          target="_blank"
        >
          <span
            className="icon"
          >
            <i
              aria-hidden="true"
              className="fab fa-github fa-lg"
            />
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
